vertx-web/src/test/java/io/vertx/ext/web/templ/impl/CachingTemplateEngineTest.java

```diff
@@ -0,0 +1,54 @@
+package io.vertx.ext.web.templ.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author <a href="https://github.com/mystdeim">Roman Novikov</a>
+ * @since 4/1/17.
+ */
+public class CachingTemplateEngineTest {
+
+  CachingTemplateEngineImpl engine;
+
+  @Before
+  public void setUp() {
+    engine = new CachingTemplateEngineImpl(".jade");
+  }
+
+  @Test
+  public void testAdjustLocation_Empty() {
+    assertEquals("templates/index.jade", engine.adjustLocation("templates/"));
+  }
+
+  @Test
+  public void testAdjustLocation_Ext() {
+    assertEquals("templates/index.jade", engine.adjustLocation("templates/.jade"));
```

Comment:
But it's common practice in web development, when we request / that means /index.template